src/nvim/statusline.c

```diff
@@ -279,7 +279,8 @@ static void win_redr_custom(win_T *wp, bool draw_winbar, bool draw_ruler)
   // There is a tiny chance that this gets called recursively: When
   // redrawing a status line triggers redrawing the ruler or tabline.
   // Avoid trouble by not allowing recursion.
-  if (entered) {
+  if (entered || skip_redr_custom_k_event) {
+    skip_redr_custom_k_event = false;
```

Comment:
Related globals `redraw_tabline` and `redraw_cmdline` look very similar in nature. Would it make sense to name this one `redraw_win_custom` ?